### DISCUSSION TITLE
USB Wifi dongle

### DISCUSSION THREAD

**Discussion by Timmy2thumbs** [xchwarze/wifi-pineapple-cloner#20] [Q&A]
MT7612U WIFI adapter into the USB port, it didn't pick it up as a WiFi interface. After some inspection it turned out that I was accidentally sent a MT7610 chip dongle instead of the 612. The pineapple did see it in the advanced config area where it lists out USB devices, but it also gave an error where it asked if the device should be assigned an interface y/n.... there's no way to input an answer. My question therefore is, is it possible to use this dongle/chipset or a different one or is it specific to the specs you listed out?

**Comment by xchwarze** [comment]
The truth is that you have to use devices that the driver is fully functional and stable. If you use a chipset other than the one in the list, it will probably not work or will be unstable.
To simplify the task I made those lists that are fixed, because most of what you try to use does not work.

**Reply by timkatta5** [reply]
I am using alfa AWUS036NHA, the gl.inet router is not reading it. Any luck on how to fix it. Also when will you launch your software for such operations

**Reply by xchwarze** [reply]
The list of dongles that work perfectly in OpenWrt is this:
https://github.com/xchwarze/wifi-pineapple-cloner/discussions/1

And here you can see the drivers installed by default:
https://github.com/xchwarze/wifi-pineapple-cloner/blob/master/tools/builder.sh#L29
"kmod-mt76x2u kmod-rt2800-usb kmod-rtl8187"

I understand that I installed all of them to make it work "out of the box". Any other chipset that is not in this list will probably not work.

**Reply by xchwarze** [reply]
https://wikidevi.wi-cat.ru/ALFA_Network_AWUS036NHA -> Chipset: Atheros AR9002U

I have no idea if this driver exists or if it exists it works for wifi auditing.
About my project, the first version will be released in a few weeks. Although some wifi modules will be released later due to the complexity.

**Reply by timkatta5** [reply]
You are right, I installed the driver through special command for the pineapple, it worked. For those who want to learn how to install alfa wifi adapter drivers:

SSH into the wifi pineapple: ssh root@172.16.42.1
Update Package List: opkg update
Install Driver: opkg install kmod-ath9k-htc

After that alfa adapters will work fine, just make sure the pineapple has an internet connection

@xchwarze I will be sending you an email regarding the software.
